lang/java/reef-applications/reef-vortex/src/main/java/org/apache/reef/vortex/driver/DefaultVortexManager.java

```diff
@@ -0,0 +1,92 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.reef.vortex.driver;
+
+import org.apache.reef.vortex.driver.tasklet.RunningExecutors;
+import org.apache.reef.vortex.driver.tasklet.Tasklet;
+import org.apache.reef.vortex.driver.tasklet.TaskletExecutor;
+
+import javax.inject.Inject;
+import java.io.Serializable;
+import java.util.*;
+import java.util.concurrent.BlockingDeque;
+import java.util.concurrent.ExecutorService;
+import java.util.concurrent.Executors;
+import java.util.concurrent.LinkedBlockingDeque;
+import java.util.logging.Level;
+import java.util.logging.Logger;
+
+/**
+ * Default implementation of VortexManager.
+ */
+public class DefaultVortexManager implements VortexManager {
+  private static final Logger LOG = Logger.getLogger(DefaultVortexManager.class.getName());
+  private final RunningExecutors runningExecutors;
+  private final BlockingDeque<Tasklet> pendingTasklets = new LinkedBlockingDeque<>();
+  private final ExecutorService schedulerThread = Executors.newSingleThreadExecutor();
+
+  @Inject
+  public DefaultVortexManager(final RunningExecutors runningExecutors) {
+    this.runningExecutors = runningExecutors;
+    schedulerThread.execute(new Runnable() {
+      @Override
+      public void run() {
+        while (!runningExecutors.isTerminated()) {
+          try {
+            final Tasklet tasklet = pendingTasklets.takeFirst(); // blocks when no tasklet exists
+            runningExecutors.launchTasklet(tasklet); // blocks when no executor exists
+          } catch (InterruptedException e) {
+            LOG.log(Level.INFO, "Interrupted upon termination");
+          }
+        }
+      }
+    });
+  }
+
+  @Override
+  public void enqueueNewTasklet(final Tasklet tasklet) {
```

Comment:
Yes it is accessed concurrently by multiple threads(the user thread, the schedulet thread and `EvaluatorMessageDispatcher` threads). Concurrency control is done inside 2 data structures(`pendingTasklets` and `runningExecutors`) it uses, so we do not have any locks or synchronized blocks in this class. 

I'll add this to the documentation.